backend/generating/SupervisionTemplate.java

```diff
@@ -0,0 +1,69 @@
+package generation;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class SupervisionTemplate{
+
+    private SupervisionInfo supervisionInfo;
+    private SupervisionQuestions questions;
+
+
+    public SupervisionTemplate(SupervisionInfo supervisionInfo, SupervisionQuestions questions) {
+        //these are the things that the user has to specify when modifying the template
+        this.supervisionInfo = supervisionInfo;
+        this.questions = questions;
+    }
+
+    public void setSupervisionHeader(){
+        LinkedHashMap<String, String> studentData = new LinkedHashMap<>();
+        studentData.put("studentname", supervisionInfo.getStudentName());
+        studentData.put("studentemail", supervisionInfo.getStudentEmail());
+        studentData.put("svrname", supervisionInfo.getSvrName());
+        studentData.put("svvenue", supervisionInfo.getSvVenue());
+        studentData.put("svcourse", supervisionInfo.getSvCourse());
+        studentData.put("svnumber", supervisionInfo.getSvNumber());
+        studentData.put("svdate", supervisionInfo.getSvDate());
+        studentData.put("svtime", supervisionInfo.getSvTime());
+        try {
+            create(studentData);

```

Comment:
Why can't the create method get this info directly from `supervisionInfo`? Is there any reason we need to copy all this data again?